src/Symfony/Component/Console/Command/Command.php

```diff
@@ -89,6 +88,8 @@ public function setApplication(Application $application = null)
         } else {
             $this->helperSet = null;
         }
+
+        $this->fullDefinition = null;
```

Comment:
All resets are BC related.. so we have a few options here;
- silently reset like now
- throw BadMethodCall if set already
- sync if set already (include arguments by design)
- dont do anything